### PR TITLE
unittest-4.1: Provided function-to-function unit-test coverage on JS tag editor and security group rules widgets

### PR DESCRIPTION
**Pull request by kyolee310**

### CODE REVIEW

**Comment by euca-qa-gui-tester** [comment]
Build triggered.

**Comment by euca-qa-gui-tester** [comment]
Merged build triggered.

**Comment by euca-qa-gui-tester** [comment]
Build started.

**Comment by euca-qa-gui-tester** [comment]
Build finished. Test PASSed.

**Comment by euca-qa-gui-tester** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com//job/user-console-PR-monitor/44/
Test PASSed.

**Comment by euca-qa-gui-tester** [comment]
Merged build started.

**Comment by euca-qa-gui-tester** [comment]
Merged build finished. Test PASSed.

**Comment by euca-qa-gui-tester** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com//job/user-console-PR-monitor/45/
Test PASSed.

File: eucaconsole/static/js/widgets/securitygroup_rules.js [hunk 1]
```diff
@@ -7,9 +7,9 @@
 angular.module('SecurityGroupRules', ['CustomFilters', 'EucaConsoleUtils'])
     .controller('SecurityGroupRulesCtrl', function ($scope, $http, $timeout, eucaUnescapeJson) {
         $http.defaults.headers.common['X-Requested-With'] = 'XMLHttpRequest';
-        $scope.rulesEditor = $('#rules-editor');
-        $scope.rulesTextarea = $scope.rulesEditor.find('textarea#rules');
-        $scope.rulesEgressTextarea = $scope.rulesEditor.find('textarea#rules_egress');
+        $scope.rulesEditor = undefined;
```
Comment (kamalgill) [inline eucaconsole/static/js/widgets/securitygroup_rules.js]: @kyolee310: Why change these to undefined?

Comment (kyolee310) [inline eucaconsole/static/js/widgets/securitygroup_rules.js]: The initialization of the jQuery objects has been moved to the init function rather than setting them as default values of the controllers.

Comment (kamalgill) [inline eucaconsole/static/js/widgets/securitygroup_rules.js]: @kyolee310: That part is clear, but *why* was that change necessary?

Comment (kyolee310) [inline eucaconsole/static/js/widgets/securitygroup_rules.js]: When writing unit-test, it was noticed that the initialization of those objects would always get triggered when the controller was loaded, which would result in error when the proper template elements were not present. This condition would prevent each function in the controller from being tested independent of the peripheral, unrelated elements in the controller. Thus, the initialization of the jQuery objects has been moved to the init function to allow the initialization of the objects to be triggered explicitly rather than automatically upon loading of the controller.
